File: entregavel/MaxDivisorComum.js
// Função que calcula o MDC de forma recursiva
function calcularMDC(a, b) {
    if (b === 0) {
        return a;
    }
    return calcularMDC(b, a % b);
}

// Testando a função
let a = 48;
let b = 18;

console.log("O MDC de " + a + " e " + b + " é: " + calcularMDC(a, b));

File: entregavel/MaxDivisorComum.java
public class MaxDivisorComum {
    public static void main(String[] args) {
        int a = 48;
        int b = 18;
        
        System.out.println("O MDC de " + a + " e " + b + " é: " + calcularMaxDivisorComum(a, b));
    }

    // Método que calcula o MDC de forma recursiva
    public static int calcularMaxDivisorComum (int a, int b) {
        if (b == 0) {
            return a;
        }
        return calcularMaxDivisorComum (b, a % b);
    }
}
